src/variorum/config_architecture.h

```diff
@@ -268,7 +282,8 @@ int variorum_exit(const char *filename,
 
 void variorum_get_topology(unsigned *nsockets,
                            unsigned *ncores,
-                           unsigned *nthreads);
+                           unsigned *nthreads,
+                           int idx);
```

Comment:
what is this index being used for?

Reply:
The number of sockets, threads etc and the hostname is set in the variorum_get_topology function in config_architecture.c, which still needs to be done per platform.

Reply:
I'm seeing that sometimes `P_MSR_CORE_IDX` is being passed into `variorum_get_topology`, what does this index represent?

Reply:
`MSR_CORE_IDX` was added for scenarios where we need to share the msr_core.c functionality -- especially among Intel and AMD CPUs. Previously, that msr_core.c was using the `P_INTEL_CPU_IDX` internally, which will not work correctly if we have an AMD CPU platform, which would need the 1P_AMD_CPU_IDX1 instead. So, I added the MSR_CORE_IDX, which can now be set based on the platform.